pennylane/math/single_dispatch.py

```diff
@@ -138,6 +152,29 @@ def _take_autograd(tensor, indices, axis=None):
 )
 
 
+def _ifft2_tf(a, s=None, axes=(-2, -1), norm=None):
+    if axes != (-2, -1):
+        raise ValueError(
+            "TensorFlow does not support passing axes; the ifft "
+            "will always be performed over the inner-most 2 dimensions."
+        )
+
+    if norm is not None:
+        raise ValueError("TensorFlow does not support the 'norm' keyword argument.")
+
+    # TensorFlow only supports FFT of complex tensors
+    if a.dtype not in [_i("tf").complex64, _i("tf").complex128]:
+        if a.dtype is _i("tf").float64:
+            a = _i("tf").cast(a, dtype=_i("tf").complex128)

```

Comment:
Sure that it is not 64 here and 128 in the next line?

Reply:
Oops 🤦

Reply:
Oh wait, no, this is what I intended.

- If the input is `float64`, I cast to `complex128` (two double precision floats, one real one imaginary) to retain the same precision.
- Otherwise, I case to `complex64` (two single-precision floats)